### PR TITLE
Updates the database

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

About time

https://github.com/tgstation/tgstation/pull/51255 and followups
https://github.com/tgstation/tgstation/pull/50843

Also fixed Midround Stickyban queries apparently failing constantly

## Changelog
:cl:
fix: Midround Stickybans will now be applied correctly
refactor: The database now uses Rustg instead of BSQL, meaning the server will crash less
config: Added missing config entries to config, as well as an option for players with enough living time to bypass Bunkers
admin: Admins will now be alerted to new players connecting
admin: Improved database voting system
server: Updated SQL schema
/:cl:

### CODE REVIEW

**Comment by Hughgent** [comment]
Save me @spookydonut!

**Comment by spookydonut** [comment]
@Hughgent @TiviPlus well lets start with it not passing CI

**Comment by TiviPlus** [comment]
CI fixed

**Review by spookydonut — DISMISSED** [review]
whoever test merges this should be fired from keyholders no matter who they are.

**Comment by spookydonut** [comment]
attn: @MrStonedOne

**Comment by optimumtact** [comment]
Does pykyzz not have alter perms to the tgmc database?

**Comment by spookydonut** [comment]
> 
> 
> Does pykyzz not have alter perms to the tgmc database?

Psy is on break and basically MIA @optimumtact

**Comment by TiviPlus** [comment]
@optimumtact @MrStonedOne  still waiting

**Comment by optimumtact** [comment]
I don't have the permissions necessary to apply procedures/triggers, until I have that I can't help you

File: SQL/database_changelog.txt [hunk 1]
```diff
@@ -1,13 +1,81 @@
...
+
+Updates and improvements to poll handling.
+Added the `deleted` column to tables 'poll_option', 'poll_textreply' and 'poll_vote' and the columns `created_datetime`, `subtitle`, `allow_revoting` and `deleted` to 'poll_question'.
+Changes table 'poll_question' column `createdby_ckey` to be NOT NULL and index `idx_pquest_time_admin` to be `idx_pquest_time_deleted_id` and 'poll_textreply' column `adminrank` to have no default.
+Added procedure `set_poll_deleted` that's called when deleting a poll to set deleted to true on each poll table where rows matching a poll_id argument.
+
+Created missing stickyban tables `stickyban_matched_cid` and `stickyban_matched_ip`, added columm `last_matched` to `stickyban_matched_ckey` and added column `deleted_ckey` to `messages` . These missing was causing queries to fail.
+
+ALTER TABLE `stickyban_matched_ckey`
+	ADD COLUMN `last_matched` TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP AFTER `first_matched`
```
Comment (MrStonedOne) [inline SQL/database_changelog.txt:23]:
```suggestion
	ADD COLUMN `last_matched` TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP AFTER `first_matched`;
```

**Review by MrStonedOne — CHANGES_REQUESTED** [review]
changes applied. please update the changelog to add the `;` so downstreams can just copy/paste the entire set of queries without errors.

**Comment by TiviPlus** [comment]
Woops meant to @spookydonut

**Pull request merged**